File: FlinkScalaCode/src/main/scala/com/wubaibao/flinkscala/code/chapter7/twophasecommit/JdbcCommonUtils.scala
package com.wubaibao.flinkscala.code.chapter7.twophasecommit

import java.sql.{Connection, DriverManager, SQLException}

/**
 * 连接Mysql - JDBC工具类
 */
case class JdbcCommonUtils() {
  val driver = "com.mysql.jdbc.Driver"
  val url = "jdbc:mysql://node2:3306/mydb?useSSL=false"
  val username = "root"
  val password = "123456"

  private var conn: Connection = _
  /**
   * 获取数据库连接对象
   */
  def getConnect: Connection = {
    if (conn == null) {
      try {
        Class.forName(driver)
        conn = DriverManager.getConnection(url, username, password)
        //设置手动提交事务
        conn.setAutoCommit(false)
      } catch {
        case e: SQLException =>
          throw new RuntimeException(e)
        case e: ClassNotFoundException =>
          throw new RuntimeException(e)
      }
    }
    conn
  }

  /**
   * 提交事务
   */
  def commit(): Unit = {
    if (conn != null) {
      try {
        conn.commit()
      } catch {
        case e: SQLException =>
          throw new RuntimeException(e)
      }
    }
  }

  /**
   * 回滚事务
   */
  def rollback(): Unit = {
    if (conn != null) {
      try {
        conn.rollback()
      } catch {
        case e: SQLException =>
          throw new RuntimeException(e)
      }
    }
  }

  /**
   * 关闭连接
   */
  def close(): Unit = {
    if (conn != null) {
      try {
        conn.close()
      } catch {
        case e: SQLException =>
          throw new RuntimeException(e)
      }
    }
  }
}

File: FlinkJavaCode/src/main/java/com/wubaibao/flinkjava/code/chapter7/twophasecommit/JdbcCommonUtils.java
package com.wubaibao.flinkjava.code.chapter7.twophasecommit;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

/**
 * 连接MySQL - JDBC工具类
 */
public class JdbcCommonUtils{
    private static String driver = "com.mysql.jdbc.Driver";
    private static String url = "jdbc:mysql://node2:3306/mydb?useSSL=false";
    private static String username = "root";
    private static String password = "123456";

    private static Connection conn ;

    /**
     * 获取数据库连接对象
     */
    public  Connection getConnect() {
        try {
            if(conn ==null){
                Class.forName(driver);
                conn = DriverManager.getConnection(url, username, password);
                //设置手动提交事务
                conn.setAutoCommit(false);
            }
        } catch (SQLException e) {
            throw new RuntimeException(e);
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        }
        return conn;
    }

    /**
     * 提交事务
     */
    public  void commit() {
        if (conn != null) {
            try {
                conn.commit();
            } catch (SQLException e) {
                throw new RuntimeException(e);
            }

        }
    }

    /**
     * 回滚事务
     */
    public  void rollback() {
        if (conn != null) {
            try {
                conn.rollback();
            } catch (SQLException e) {
                throw new RuntimeException(e);
            }

        }
    }

    /**
     * 关闭连接
     */
    public void close() {
        if (conn != null) {
            try {
                conn.close();
            } catch (SQLException e) {
                throw new RuntimeException(e);
            }

        }
    }
}
